### PR TITLE
Add auth info to ProjectInfo

### PR DESCRIPTION
**Pull request by albscui**
ProjectInfo is now aware of auth info when auth is active.

Pachctl can take advantage of this:

```
pachctl list projects
ACTIVE PROJECT      ACCESS_LEVEL                 DESCRIPTION
       aliceProject [clusterAdmin]               -
       otherProject [clusterAdmin]               -
*      default      [clusterAdmin projectWriter] -
```

Note the change in pfs.proto from `RepoAuthInfo` to `AuthInfo` should be backwards compatible with old clients because the proto wire format only cares about the field number, wire type, and payload, which in this case are unchanged.

### CODE REVIEW

**Review by brendoncarroll — APPROVED** [review]
PFS changes look good.  Didn't look too closely at auth stuff.

File: src/server/auth/server/testing/auth_test.go [hunk 1]
```diff
@@ -2861,3 +2861,44 @@ func TestListPipelinesWithProjectAccessControl(t *testing.T) {
...
+			adminProject:           {Roles: []string{}},
+			aliceProject:           {Roles: []string{auth.ProjectOwnerRole}},
+		}},
+	}
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			projectInfos, err := tc.client.ListProject()
+			require.NoError(t, err)
+			for _, projectInfo := range projectInfos {
+				if expected, ok := tc.expected[projectInfo.Project.Name]; ok {
```
Comment (acohen4) [inline src/server/auth/server/testing/auth_test.go:2898]: we should error if the project name is not in expected right?

**Review by acohen4 — APPROVED** [review]
LGTM - left a few nits

File: src/server/pfs/server/driver.go [hunk 2]
```diff
@@ -660,11 +660,21 @@ func (d *driver) createProjectInTransaction(txnCtx *txncontext.TransactionContex
 	return nil
 }
 
-func (d *driver) inspectProject(ctx context.Context, project *pfs.Project) (*pfs.ProjectInfo, error) {
+func (d *driver) inspectProject(ctx context.Context, project *pfs.Project, includeAuth bool) (*pfs.ProjectInfo, error) {
```
[the remark this answers is not in the document]
Comment (albscui) [reply src/server/pfs/server/driver.go:663]: No, I was mainly trying to keep parity with `inspectRepo`. I guess for now we can remove it, and add it later if necessary.

**Pull request merged**